src/main/java/com/github/tomakehurst/wiremock/admin/model/SnapshotRepeatedRequestHandler.java

```diff
@@ -0,0 +1,84 @@
+package com.github.tomakehurst.wiremock.admin.model;
+
+import com.github.tomakehurst.wiremock.common.UniqueFilenameGenerator;
+import com.github.tomakehurst.wiremock.matching.RequestPattern;
+import com.github.tomakehurst.wiremock.stubbing.Scenario;
+import com.github.tomakehurst.wiremock.stubbing.StubMapping;
+import org.apache.commons.collections4.MultiSet;
+import org.apache.commons.collections4.multiset.HashMultiSet;
+
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+
+/**
+ * Counts unique RequestPatterns from StubMappings. If shouldRecordRepeatsAsScenarios is enabled, then multiple
+ * identical requests will be recorded as scenarios. Otherwise, they're removed.
+ */
+public class SnapshotRepeatedRequestHandler {
+    private final static String SCENARIO_NAME_PREFIX = "scenario";
+    private final boolean shouldRecordRepeatsAsScenarios;
+
+    public SnapshotRepeatedRequestHandler(boolean shouldRecordRepeatsAsScenarios) {
+        this.shouldRecordRepeatsAsScenarios = shouldRecordRepeatsAsScenarios;
+    }
+
+    public void filterOrCreateScenarios(Iterable<StubMapping> stubMappings) {
```

Comment:
The reason this doesn't use `filter()` is because it's stateful (it has to keep track of the previously-seen requests), and predicates are meant to be free of side effects.